src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -0,0 +1,101 @@
+'''Defines a CoReSyF Tool consisting of a SNAP single gpt operation.'''
+import os
+import subprocess
+import re
+from tool import CoReSyFTool
+from manifest import InvalidManifestException
+
+class TooManyInputArgumentsException(Exception):
+    pass
+
+class ToManyOutputArgumentsException(Exception):
+    pass
+
+class GPTExecutionException(Exception):
+    '''Error occurred during a SNAP gpt execution'''
+
+    ERROR_REGEX = re.compile('Error:')
+
+    def __init__(self, returncode, msg):
+        self.returncode = returncode
+        self.errors = [line[7:] for line in msg.split(
+            os.linesep) if self.ERROR_REGEX.match(line)]
+        super(GPTExecutionException, self).__init__(self.errors)
+
+
+class GPTGraphFileNotFound(Exception):
+    pass
+
+
+class GPTCoReSyFTool(CoReSyFTool):
+    '''CoReSyF Tool consisting of a single SNAP gpt operation.'''
+
+    DEFAULT_FORMAT = 'GeoTIFF-BigTIFF'
+    DEFAULT_EXT = 'tif'
+    DEFAULT_GPT_GRAPH_FILE_NAME = 'gpt_graph.xml'
+
+    def _validate_operation(self, operation):
```

Comment:
This method is not used.

Reply:
It is, by the super-class. `CoReSyFTool` expects validation behavior to be defined here, so that sub-classes can extend it. The method is called during the object `__init__`.